core/src/pow/common.rs

```diff
@@ -135,31 +113,29 @@ macro_rules! to_edge {
 
 /// Utility struct to calculate commonly used Cuckoo parameters calculated
 /// from header, nonce, edge_bits, etc.
-pub struct CuckooParams<T>
-where
-	T: EdgeType,
-{
+pub struct CuckooParams {
 	pub edge_bits: u8,
 	pub proof_size: usize,
 	pub num_edges: u64,
 	pub siphash_keys: [u64; 4],
-	pub edge_mask: T,
+	pub edge_mask: u64,
+	pub node_mask: u64,
 }
 
-impl<T> CuckooParams<T>
-where
-	T: EdgeType,
-{
+impl CuckooParams {
 	/// Instantiates new params and calculate edge mask, etc
-	pub fn new(edge_bits: u8, proof_size: usize) -> Result<CuckooParams<T>, Error> {
-		let num_edges = (1 as u64) << edge_bits;
-		let edge_mask = to_edge!(T, num_edges - 1);
```

Comment:
We can remove the `to_edge` macro up now I think as it is no longer used.
And are the other `to_u64`, `to_u32` and `to_usize` macros still in use? I suspect we no longer need these now that we are doing less conversions.